htdocs/adherents/subscription.php

```diff
@@ -810,54 +847,103 @@
 		dol_fiche_head('');
 
 		print "<table class=\"border\" width=\"100%\">\n";
-        print '<tbody>';
+             print '<tbody>';
 
 		$today=dol_now();
         $datefrom=0;
         $dateto=0;
         $paymentdate=-1;
 
+$year = strftime("%Y",$today);
+$month = strftime("%m",$today);
+$day = strftime("%d",$today);
+
+if ($object->datefin== NULL){
+$datefin=dol_now()-86400;
+}else {
+$datefin=$object->datefin+86400;
+}
+
+$cotisationfin = strftime("%Y",$datefin);
+$cotisationfin2 = strftime("%m",$datefin);
+$cotisationfin3 = strftime("%d",$datefin);
+
+$startcotis0 = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,'01',$year-1);
+$startcotis1 = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,'01',$year);
+$startcotis2 = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,'01',$year+1);
+
+if ($startcotis1>$today){
+if ($conf->global->ADHERENT_SUBSCRIPTION_PRORATA == '0') { 
+$next = mktime(5,00,00,$month-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,$day,$year);
+$date = $dateb = $today;
+}else{
+$next = $startcotis1;
+if (dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year-1)>$today && $datefin<$today){$date=dol_now();}else{$date = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year-1);}
+$dateb = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year-1);}
+$dateto = strtotime(date("Y-m-d", $dateb) . " + 1 year - 1 day");  
+}else{
+if ($conf->global->ADHERENT_SUBSCRIPTION_PRORATA == '0') {
+$next = mktime(0,00,00,$month-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,$day,$year+1);
+$date = $dateb =$today;
+}else{ 
+$next = $startcotis2; 
+if (dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year)>$today && $datefin<$today){$date=dol_now();}else{$date = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year);}
+$dateb = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year);} 
+$dateto = strtotime(date("Y-m-d", $dateb) . " + 1 year - 1 day");
```

Comment:
use the method   dol_timeplusduree() to make addition and substraction on date.

Reply:
@eldy fix done ;-)